transformer/transformers/ampruntimejs.go

```diff
@@ -0,0 +1,35 @@
+package transformers
+
+import (
+	"net/url"
+	"strings"
+
+	"github.com/ampproject/amppackager/transformer/internal/amphtml"
+	"github.com/ampproject/amppackager/transformer/internal/htmlnode"
+	"golang.org/x/net/html"
+	"golang.org/x/net/html/atom"
+)
+
+// AMPRuntimeJS rewrites the value of src in script nodes, where applicable.
+// If the value is of the form "*.js", replace it with "*.js?f=sxg".
+func AMPRuntimeJS(e *Context) error {
+	for n := e.DOM.HeadNode; n != nil; n = htmlnode.Next(n) {
```

Comment:
Continuing https://github.com/ampproject/amppackager/pull/451/files#r469658501 (I can't unresolve it for some reason):

> Replacing with `n.NextSibling` actually skipped over the transformation altogether

The other test cases are failing because they are not in a `<head>`.

> `htmlnode.Next` passes both

The issue isn't that `Next` won't work, it's that it's doing **too much** work. The scripts that we're concerned with are only valid inside a `<head>` node. However, `Next` will traverse the `<head>` and then everything inside the `<body>` (which can be a lot of nodes). I'm confident `NextSibling` is the right code here.

Reply:
Here are the results of the two failed tests using `NextSibling`.
```
ampruntimejs_test.go:72: transformation: Transform=
        "<html><head><script async=\"\" src=\"https://cdn.ampproject.org/v0.js\"></script></head><body></body></html>"
        want=
        "<html><head><script async=\"\" src=\"https://cdn.ampproject.org/v0.js?f=sxg\"></script></head><body></body></html>"
    ampruntimejs_test.go:72: transform on two scripts: Transform=
        "<html><head><script async=\"\" src=\"https://cdn.ampproject.org/foo.js\"></script><script async=\"\" src=\"https://cdn.ampproject.org/bar.js\"></script></head><body></body></html>"
        want=
        "<html><head><script async=\"\" src=\"https://cdn.ampproject.org/foo.js?f=sxg\"></script><script async=\"\" src=\"https://cdn.ampproject.org/bar.js?f=sxg\"></script></head><body></body></html>"
```
As you can see, these tests fail, and they're both in a `<head>`. The other two tests that are not in `<head>` actually passes.
Maybe there is a better way to do this? Such as starting at the first child of the `HeadNode`?